### DISCUSSION TITLE
How to change the listening, serving & advertising URL from localhost to my system's actual IP?

### DISCUSSION THREAD

**Discussion by dev-vaayen** [etcd-io/etcd#16524] [Q&A] [answered]
Hi all!

I have installed etcd on my system (without containerisation) using the etcd document (https://etcd.io/docs/v3.5/install/).

The listening ,serving and advertising URL are set as localhost by default. Can you please tell me how I can change it from localhost to my actual IP address?

Are there any changes that I can do to some config file for this?

**Comment by jmhbnz** [comment, marked as answer]
Hey @TheUncertainOracle - Thanks for your question, please refer to the etcd configuration options for changing these values: https://etcd.io/docs/v3.5/op-guide/configuration

You can configure these options through:
- Environment variables
- Command line parameters
- [Configuration file](https://github.com/etcd-io/etcd/blob/main/etcd.conf.yml.sample)

The parameters I believe you are looking for are:

```bash
--advertise-client-urls 'http://localhost:2379'
  List of this members client URLs to advertise to the public.
--listen-client-urls 'http://localhost:2379'
  List of URLs to listen on for client traffic.
--listen-peer-urls 'http://localhost:2380'
  List of URLs to listen on for peer traffic.
--initial-advertise-peer-urls 'http://localhost:2380'
  List of this members peer URLs to advertise to the rest of the cluster.
```
